### DISCUSSION TITLE
For loop with multiple blocks with  synchronous input

### DISCUSSION THREAD

**Discussion by askherconsulting** [postmanlabs/postman-flows#329] [Q&A] [answered]
Question posted by Josh on my blog:

My use-case: I have many different customer accounts (username/password) with different configurations/permissions. I want to send each account through the flow one-at-a-time; so Account 2 would only begin after Account 1 has completed the Flow. This ensures that my Flow works for all the different account configurations/permissions. Do you have any examples of how I can do a multi-block 'For Each'/'Loop' with a mechanism that only allows 1 account in the 'For Each' loop at a time?

Please provide an example if possible and I will feed this back to Josh.

**Comment by samay-v** [comment, marked as answer]
Hey @askherconsulting 
these are a few example flows that can help Josh:
1. [Looping over Lists](https://www.postman.com/postman/workspace/example-flows/flow/6267f947d0fb400392442ac9)
2. [Persisting pagination](https://www.postman.com/postman/workspace/example-flows/flow/629999204373b20e633bf1e7)

I think the concept of [Signals](https://learning.postman.com/labs/postman-flows/core-concepts/connections/#signals) will also come in handy. 

If Josh has any further doubts, I think understanding his usecase better would help. If they can put the flow in a public workspace and share, that could also be great

**Reply by productman-josh** [reply]
Thanks samay-v! The 'Persisting pagination' example flow gives me a lot of possible solutions to my problem. As a Product Manager- thank you for the great product! I use Postman heavily during my workday to explore business rules with APIs. I'm excited to have our entire company adopt a Flows-first mentality for communicating business rules once Flows has a few more tools to make something like Function Flows that are callable/reusable. Flows is really impressive and has a big impact on my life-- thanks again for being involved in something that's helpful in aiding communication between the product business rules, QA, and the engineering team. 
-Josh

**Reply by samay-v** [reply]
@productman-josh Thanks for that encouraging feedback! 
Very excited to see you're using flows to communicate Business Rules. While we work on adding the features you talk about, I'd love to get to know more about how you and your team is using flows. If you'd like to get in touch, hit me up at samay.verma[at]postman.com
